src/util/usertools.c

```diff
@@ -572,10 +573,23 @@ sss_fqname(char *str, size_t size, struct sss_names_ctx *nctx,
 
 errno_t sss_user_by_name_or_uid(const char *input, uid_t *_uid, gid_t *_gid)
 {
+    static struct sss_nss_ops nss_ops;
     uid_t uid;
     errno_t ret;
     char *endptr;
-    struct passwd *pwd;
+    struct passwd pwd = { 0 };
+    int errnop = 0;
+    enum nss_status status;
+    char s_nss_buffer[NSS_BUFFER_SIZE];
```

Comment:
I usually use `s_` prefix to indicate static variables.

Perhaps this is way too old-fashioned, but imo it indeed makes some sense to make this buffer static to avoid allocating 16kb on the stack.